### DISCUSSION TITLE
Loading backbone (resnet50) from locally mounted directory

### DISCUSSION THREAD

**Discussion by Tobias1234** [azavea/raster-vision#1488] [Q&A] [answered]
I am trying to use Raster vision with Docker on a host computer using a corporate network . 
When the python script tries to download the resnet model from "https://download.pytorch.org/models/resnet50-19c8e357.pth" it fails with  "urlopen error [SSL: CERTIFICATE_VERIFY_FAILED]"
SSL errors seems quite tricky so solve so my workaround would be to download the model on the host and put it in a mounted map, and change the code to the local directory `model=SemanticSegmentationModelConfig(backbone='/opt/src/code/resnet50.pth'),`

But I get an error:   
```
model=SemanticSegmentationModelConfig(backbone='/opt/src/code/resnet50.pth'),
  File "pydantic/main.py", line 274, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for SemanticSegmentationModelConfig
backbone
  value is not a valid enumeration member; permitted: 'alexnet', 'densenet121', 'densenet169', 'densenet201', 'densenet161', 'googlenet', 'inception_v3', 'mnasnet0_5', 'mnasnet0_75', 'mnasnet1_0', 'mnasnet1_3', 'mobilenet_v2', 'resnet18', 'resnet34', 'resnet50', 'resnet101', 'resnet152', 'resnext50_32x4d',....
```
Is it possible to load backbone local, something like the way pretrained weights can be loaded locally?

**Comment by AdeelH** [comment, marked as answer]
One simple workaround is to put the file in the directory where it tries to download it since it doesn't re-download if the file already exists. I believe the correct path would be `/opt/data/torch-cache/hub/checkpoints/resnet50-19c8e357.pth` or possibly `/root/.cache/torch/hub/checkpoints/resnet50-19c8e357.pth`.

**Reply by Tobias1234** [reply]
Running on my personal computer I never mounted to or seen the folder "torch-cache" locally and every time I run a pythonscript the resnet model is downloaded. But I guess by creating a local folder called "torch-cache" mounting it in Docker and putting the subdirectory hub/checkpoints/resnet50-19c8e357.pth in it will solve the problem.

**Reply by AdeelH** [reply]
You can also change the download dir by setting the `TORCH_HOME` env variable. See https://pytorch.org/docs/stable/hub.html#where-are-my-downloaded-models-saved.

**Reply by Tobias1234** [reply]
I tried using using the _-e_ flag to change the environment variable (Docker on Windows). Maybe I did something wrong, but instead I solved it by creating a folder "RV_PYTHON_DIR" on local disc and putting the resnet file there and then using `docker run --mount type=bind,source="C:/Users/tt742/RV2/RV_PYTHON_DIR",target=/opt/data/torch-cache/hub/checkpoints/ ` This worked as workaround!
